### ISSUE TITLE
[Woo POS] [Variable Products] Add new in-line error state for variations and products

### ISSUE THREAD

**Issue by joshheald** [woocommerce/woocommerce-ios#14694]
## Description

Our existing error state for loading items is primarily designed for errors when loading the initial page of results, which is what it was built for.

If there's an error, we show a full screen error, with a retry button, which hides any previously loaded items. This includes when loading a second page of the list. 

This approach already makes little sense, but it'll be even stranger if we keep this approach when we support variations.

For this reason, we're going to add a new in-line error state.

![Product list with inline error shown as a row at the bottom of the list](https://github.com/user-attachments/assets/ff79e72b-4964-41d3-a6ee-e12d6263bf57)

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by jaclync** [comment]
Nice, yea it doesn't make sense to show a fullscreen UI for error loading variations. Just to clarify:

- Is the error a row at the end of the existing variation list? If so, do we need to scroll to the bottom if the scroll position isn't at the bottom?
- What about a banner with full width fixed at the bottom in the item selector view?

**Comment by joshheald** [comment]
@jaclync Yeah, good point, if we get the error on pull to refresh, it's not much use as a row at the bottom of the list. 

There are a few designs here: rNAclWFAdQRw7EXEcsNVai-fi-3217_4277, the one in the screenshot is the one Android are using. I think the main use case that's been considered is an error when loading the second page or beyond.

Having an overlay at the bottom, instead of a row or background at the bottom, will work for both cases. Alternatively, we could insert the row at the bottom for an error after a new page load, or at the top for an error when pulling to refresh.
